doc/man3/EVP_PKEY_verify.pod

```diff
@@ -2,59 +2,165 @@
+followed by EVP_PKEY_verify_final() call.
 
 =head1 NOTES
 
-After the call to EVP_PKEY_verify_init() algorithm specific control
-operations can be performed to set any appropriate parameters for the
-operation.
+=begin comment
+
+These notes are largely replicated in EVP_PKEY_sign.pod, please keep them
+in sync.
+
+=end comment
+
+=head2 General
+
+Some signature implementations only accumulate the input data and do no
+further processing before verifying it (they expect the input to be a digest),
+while others compress the data, typically by internally producing a digest,
+and signing the result, which is then verified against a given signature.
+Some of them support both modes of operation at the same time.
+The caller is expected to know how the chosen algorithm is supposed to behave
+and under what conditions.
+
+For example, an RSA implementation can be expected to only expect a digest as
+input, while ED25519 can be expected to process the input with a hash, i.e.
+to produce the digest internally, and while RSA-SHA256 can be expected to
+handle either mode of operation, depending on if the operation was initialized
+with EVP_PKEY_verify_init_ex2() or with EVP_PKEY_verify_message_init().
+
+Similarly, an RSA implementation usually expects additional details to be set,
+like the message digest algorithm that the input is supposed to be digested
+with, as well as the padding mode (see L<EVP_PKEY_CTX_set_signature_md(3)> and
+L<EVP_PKEY_CTX_set_rsa_padding(3)> and similar others), while an RSA-SHA256
+implementation usually has these details pre-set and immutable.
 
-The function EVP_PKEY_verify() can be called more than once on the same
-context if several operations are performed using the same parameters.
+The functions described here can't be used to combine separate algorithms.  In
+particular, neither L<EVP_PKEY_CTX_set_signature_md(3)> nor the B<OSSL_PARAM>
+parameter "digest" (B<OSSL_SIGNATURE_PARAM_DIGEST>) can not be used to combine
+a signature algorithm with a hash algorithm to process the input.  In other
+words, it's not possible to specify a I<ctx> pre-loaded with an RSA pkey, or
+an I<algo> that fetched C<RSA> and try to specify SHA256 separately to get the
+functionality of RSA-SHA256.  If combining algorithms in that manner is
+desired, please use L<EVP_DigestVerifyInit(3)> and associated functions, or
+L<EVP_VerifyInit(3)> and associated functions.
+
+=head2 Performing multiple signatures
+
+EVP_PKEY_verify_message_final() resets the context I<ctx> in such a way that
+another sequence of (multiple) EVP_PKEY_verify_update() and
+EVP_PKEY_verify_message_final() can be performed using the parameters that was
+given by the initializing call.
+Likewise, EVP_PKEY_verify() can be called more than once on the same context to
+have several one-shot operations are performed using the same parameters.
+
+=head2 On EVP_PKEY_CTX_set_signature()
+
+Signature algorithms have emerged that require the signature to be verified
```

Comment:
Maybe..


Some signature algorithms (such as LMS) require the signature verification data be specified before verifying the message.

Reply:
It's exactly LMS that I have in mind here.  But, my phrasing may be off, please help make it better?

Reply:
Above was my suggestion :)

Reply:
Ahhhhh!  Ok, sorry that I didn't catch that

Reply:
Updated